rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -107,40 +107,40 @@ std::shared_ptr<PublisherData> PublisherData::make(
     return nullptr;
   }
 
-  zenoh::ZResult result;
-  std::optional<zenoh::ext::PublicationCache> pub_cache;
-  zenoh::KeyExpr pub_ke(entity->topic_info()->topic_keyexpr_);
-  // Create a Publication Cache if durability is transient_local.
-  if (adapted_qos_profile.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL) {
-    zenoh::ext::SessionExt::PublicationCacheOptions pub_cache_opts =
-      zenoh::ext::SessionExt::PublicationCacheOptions::create_default();
-
-    pub_cache_opts.history = adapted_qos_profile.depth;
-    pub_cache_opts.queryable_complete = true;
+  using AdvancedPublisherOptions = zenoh::ext::SessionExt::AdvancedPublisherOptions;
+  auto adv_pub_opts = AdvancedPublisherOptions::create_default();
 
-    std::string queryable_prefix = entity->zid();
-    pub_cache_opts.queryable_prefix = zenoh::KeyExpr(queryable_prefix);
-
-    pub_cache = session->ext().declare_publication_cache(
-      pub_ke, std::move(pub_cache_opts), &result);
+  if (adapted_qos_profile.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL) {
+    // Allow this publisher to be detected through liveliness.
+    adv_pub_opts.publisher_detection = true;
+  }
 
-    if (result != Z_OK) {
-      RMW_SET_ERROR_MSG("unable to create zenoh publisher cache");
-      return nullptr;
-    }
+  if (adapted_qos_profile.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL ||
+    adapted_qos_profile.reliability == RMW_QOS_POLICY_RELIABILITY_RELIABLE)
+  {
+    adv_pub_opts.cache = AdvancedPublisherOptions::CacheOptions::create_default();
+    adv_pub_opts.cache->max_samples = adapted_qos_profile.depth;
```

Comment:
Based on this logic, a PubCache is created even if durability is volatile as long as reliability is reliable. Isn't this a change in behavior? Why do we need to check for reliability here?

Reply:
It was needed if "end-to-end" reliability is required. But since we have decided to exclude it from this PR, let's roll back to its original setting.

Reply:
Sounds good, I pushed some minor changes in https://github.com/ros2/rmw_zenoh/pull/368/commits/15d522a8abe6f183999e4942febd77c19f160703